src/state/lottery/getUserTicketsData.ts

```diff
@@ -0,0 +1,69 @@
+import { TICKET_LIMIT_PER_REQUEST } from 'config/constants/lottery'
+import { LotteryTicket } from 'config/constants/types'
+import { UserTicketsResponse } from 'state/types'
+import { getLotteryV2Contract } from 'utils/contractHelpers'
+
+const lotteryContract = getLotteryV2Contract()
+
+export const processRawTicketsResponse = (ticketsResponse: UserTicketsResponse): LotteryTicket[] => {
+  const [ticketIds, ticketNumbers, ticketStatuses] = ticketsResponse
+
+  if (ticketIds?.length > 0) {
+    return ticketIds.map((ticketId, index) => {
+      return {
+        id: ticketId.toString(),
+        number: ticketNumbers[index].toString(),
+        status: ticketStatuses[index],
+      }
+    })
+  }
+  return []
+}
+
+export const viewUserInfoForLotteryId = async (
+  account: string,
+  lotteryId: string,
+  cursor: number,
+  perRequestLimit: number,
+): Promise<LotteryTicket[]> => {
+  try {
+    const data = await lotteryContract.viewUserInfoForLotteryId(account, lotteryId, cursor, perRequestLimit)
+    return processRawTicketsResponse(data)
+  } catch (error) {
+    console.error('viewUserInfoForLotteryId', error)
+    return null
+  }
+}
+
+export const fetchUserTicketsForOneRound = async (account: string, lotteryId: string): Promise<LotteryTicket[]> => {
```

Comment:
Function that will return all user tickets for one round. It keeps making node calls until fewer than 2500 tickets (`TICKET_LIMIT_PER_REQUEST`) are returned